### DISCUSSION TITLE
Is new timestamp behavior due to geopandas 1.0 or a different library?

### DISCUSSION THREAD

**Discussion by Erotemic** [geopandas/geopandas#3354] [Q&A] [answered]
The code:

```
    import pathlib
    fpath = pathlib.Path('foo.json')
    fpath.write_text('{"type": "FeatureCollection", "features": [{"type": "Feature", "properties": {"start_date": "2011-01-26"}}]}')
    gpd.read_file(fpath)['start_date']
```

Is working differently in geopandas 1.0 compared to 0.14.4. It is now automagically converting the json string to a `datetime64[ms]` column. Previously it was an object column. Is the change in geopandas itself, some new default, or is there a way to get the old behavior? My code is now breaking because the rows are no longer json serializable.

**Comment by jorisvandenbossche** [comment, marked as answer]
This is a consequence of the change of the default IO engine from fiona to pyogrio.

As a quick workaround, you can specify the `engine` manually to get the old behaviour:

```python
>>> geopandas.read_file(fpath)['start_date']  # default is engine="pyogrio"
0   2011-01-26
Name: start_date, dtype: datetime64[ms]

>>> geopandas.read_file(fpath, engine="fiona")['start_date']
0    2011-01-26
Name: start_date, dtype: object
```

Now, I was a bit surprised to see this behaviour change, because we are testing that also with fiona you get datetime64 values when roundtripping such data to GeoJSON. 
But, it seems this issue is specific to dates, i.e. without a time component. If I tweak your example slightly to use "2011-01-26T09:00" instead of "2011-01-26" in the json file, then both fiona and pyogrio give  datetime64 dtype:

```python
fpath2 = pathlib.Path('foo2.json')
fpath2.write_text('{"type": "FeatureCollection", "features": [{"type": "Feature", "properties": {"start_date": "2011-01-26T09:00"}}]}')

>>> geopandas.read_file(fpath2, engine="fiona")['start_date']
0   2011-01-26 09:00:00
Name: start_date, dtype: datetime64[ms]
```

And in our code for reading with fiona, we only handle fields with a type of "datetime" to parse them to a datetime64 dtype. But for the original example here we get a field with type "date":

```
>>> with fiona.open(fpath) as features:
...     print(features.schema)
{'properties': {'start_date': 'date'}, 'geometry': 'Unknown'}

>>> with fiona.open(fpath2) as features:
...    print(features.schema)
{'properties': {'start_date': 'datetime'}, 'geometry': 'Unknown'}
```

**Reply by jorisvandenbossche** [reply]
As for the question if there is a way to get the old behaviour, but with keep using `pyogrio`, I don't think pyogrio currently exposes anything to control that _directly_.

However, if you specify to use the Arrow-based reader interface (`use-arrow=True`, this takes a different, faster code path in pyogrio but that is only available for recent GDAL and when pyarrow is installed), then it actually gives you python `datetime.date` objects (not strings as with fiona, but the datetime.date objects might be json serializable?)

```python
>>> geopandas.read_file(fpath, engine="pyogrio", use_arrow=True)['start_date']
0    2011-01-26
Name: start_date, dtype: object

>>> geopandas.read_file(fpath, engine="pyogrio", use_arrow=True, arrow_to_pandas_kwargs=dict(date_as_object=False))['start_date']
0   2011-01-26
Name: start_date, dtype: datetime64[ms]
```

As another workaround, you can of course always convert the datetime64 column to strings after reading ..
